src/components/submissionPage/SubmissionTextBox.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useState, useEffect } from 'react';
+import Snap from '../../firebase/snap/Snap';
+import SubmitSnap from '../../firebase/snap/SubmitSnap';
+
+
+const SubmissionTextBox:React.FunctionComponent = () => {
+  /* Containing body of the snap */
+  const [message, setMessage] = useState<String>("");
+  const [charactersLeft, setCharactersLeft] = useState<Number>(280);
```

Comment:
I don't think you need _state_ for this. It's not an expensive value to calculate so we should derive it instead.